File: solutions/java/src/stackoverflow/Tag.java
package stackoverflow;

import java.util.UUID;

public class Tag {
    private final String id;
    private final String name;

    public Tag(String name) {
        this.id = UUID.randomUUID().toString();
        this.name = name;
    }

    // Getters
    public String getId() { return id; }
    public String getName() { return name; }
}

File: solutions/typescript/src/StackOverflow/Tag.ts
import { v4 as uuidv4 } from "uuid";

export default class Tag {
    private id: string;
    private name: string;

    constructor(name: string) {
        this.id = uuidv4();
        this.name = name;
    }

    getId() {
        return this.id;
    }
    getName() {
        return this.name;
    }
}
